src/app/(pages)/educations/[id]/_components/EducationsForm.tsx

```diff
@@ -0,0 +1,206 @@
+import { useForm } from "react-hook-form";
+import FormInput from "~/components/organisms/form/formInput/FormInput";
+import FormSelect from "~/components/organisms/form/formSelect/FormSelect";
+import FormTextarea from "~/components/organisms/form/formTextArea/FormTextArea";
+import { Button } from "~/components/atoms/Button";
+import FormComponent from "~/components/organisms/form/formComponent/FormComponent";
+import React, { useEffect, useState } from "react";
+import {
+  translateEducationType,
+  mapTranslatedEducationType,
+} from "~/app/(pages)/educations/utils";
+import { EducationType } from "~/server/db/schema";
+import { api } from "~/trpc/react";
+import { useRouter } from "next/navigation";
+
+type EducationFormData = {
+  type: string;
+  title: string;
+  description: string;
+  precondition: string;
+  duration: string;
+  lecturers: string;
+  courseDuration: string;
+  renewalDuration: string;
+  topics: string;
+};
+
+type Props = {
+  id: string;
+  formData: EducationFormData;
+};
+
+const EducationForm: React.FC<Props> = ({ id, formData }) => {
+  const [uniqueTypes, setUniqueTypes] = useState<string[]>([]);
+  const form = useForm<EducationFormData>({
+    defaultValues: {
+      type: formData.type,
+      title: formData.title,
+      description: formData.description,
+      precondition: formData.precondition,
+      duration: formData.duration,
+      lecturers: formData.lecturers,
+      courseDuration: formData.courseDuration,
+      renewalDuration: formData.renewalDuration,
+      topics: formData.topics,
+    },
+  });
+  const router = useRouter();
+
+  const { isSubmitting } = form.formState;
+  const createEducation = api.education.create.useMutation();
+  const updateEducation = api.education.update.useMutation();
+  const getUniqueTypes = api.education.getUniqueTypes.useQuery();
+
+  useEffect(() => {
+    if (getUniqueTypes.data) {
+      setUniqueTypes(getUniqueTypes.data.map((type) => type.type));
+    }
+  }, [getUniqueTypes.data]);
+
+  // Handle form submission
+  const handleSubmit = async () => {
+    try {
+      const inputtedType = form.getValues("type");
+      const inputtedTitle = form.getValues("title");
+      const inputtedDescription = form.getValues("description");
+      const inputtedPrecondition = form.getValues("precondition");
+      const inputtedDuration = form.getValues("duration");
+      const inputtedLecturers = form.getValues("lecturers");
+      const inputtedCourseDuration = form.getValues("courseDuration");
+      const inputtedRenewalDuration = form.getValues("renewalDuration");
+      const inputtedTopics = form.getValues("topics");
+
+      let translatedType = mapTranslatedEducationType(inputtedType) as string;
+      if (mapTranslatedEducationType(inputtedType) === undefined) {
```

Comment:
It would be possible if the type wasn't one of the 3 types we have defined. But as I see now that our db schema doesn't allow other types, it makes sense to remove this